kramer/src/test/kotlin/kramer/integration/KramerIntegrationTest.kt

```diff
@@ -0,0 +1,177 @@
+ * repositories. That said, from a home office internet it completes in less than 10 seconds
+ * on average (which includes 2 large no-cache scenarios).
+ */
+
+class KramerIntegrationTest {
+  private val relativeDir = "test_workspace/src/test/kotlin"
+  private val packageDir = this.javaClass.`package`!!.name.replace(".", "/")
+  private val tmpDir = Files.createTempDirectory("resolution-test-")
+  private val cacheDir = tmpDir.resolve("localcache")
+  private val runfiles = Paths.get(System.getenv("JAVA_RUNFILES")!!)
+  val repoArgs = listOf(
+    "--repository=${MAVEN_CENTRAL.id}|${MAVEN_CENTRAL.url}",
+    "--repository=${GOOGLE_ANDROID.id}|${GOOGLE_ANDROID.url}",
+    "--repository=spring_io_plugins|https://repo.spring.io/plugins-release",
+    "--local_maven_cache=$cacheDir"
+  )
+  private val baos = ByteArrayOutputStream()
+  private val cmd = Kramer(output = PrintStream(baos)).subcommands(ResolveOne(), MavenRepo())
+
+  @After fun tearDown() {
+    cacheDir.toFile().deleteRecursively()
+    check(!Files.exists(cacheDir)) { "Failed to tear down and delete temp directory." }
+  }
+
+  @Test fun simpleResolution() {
+    val args = configFlags("simple", "gen-maven-repo")
+    val output = cmd.test(args)
+    assertThat(output).contains("Building workspace for 1 artifacts")
+    assertThat(output).contains("Writing 1 build files into workspace")
+    assertThat(output).contains("Resolved 1 artifacts with 100 threads in")
+  }
+
+  @Test fun excludesSuccess() {
+    val args = configFlags("excludes-success", "gen-maven-repo")
+    val output = cmd.test(args)
+    assertThat(output).contains("Building workspace for 1 artifacts")
+    assertThat(output).contains("Writing 1 build files into workspace")
+    assertThat(output).contains("Resolved 1 artifacts with 100 threads in")
+  }
+
+  @Test fun excludesFailure() {
+    val args = configFlags("excludes-failure", "gen-maven-repo")
+    val output = cmd.fail(args)
+    assertThat(output).contains("Building workspace for 1 artifacts")
+    assertThat(output).contains("Writing 1 build files into workspace")
+    assertThat(output).contains("Resolved 1 artifacts with 100 threads in")
+
+    assertThat(output)
+      .contains("ERROR: Un-declared artifacts referenced in the dependencies of some artifacts.")
+    assertThat(output)
+      .contains(""""org.apache.maven:maven-builder-support:3.6.3": {"insecure": True}""")
+    assertThat(output).contains(""""exclude": ["org.apache.maven:maven-builder-support"]""")
+  }
+
+  @Test fun largeListOfArtifacts() {
+    val args = configFlags("large", "gen-maven-repo")
+    val output = cmd.test(args)
+    assertThat(output).contains("Building workspace for 468 artifacts")
+    assertThat(output).contains("Writing 229 build files into workspace")
+    assertThat(output).contains("Resolved 468 artifacts with 100 threads in")
+  }
+
+  // This is the flakiest test design that ever flaked, but we want a sense that there is
```

Comment:
Add @Ignore. Don't leave the entire block commented out -- leads to faster bitrot.

Reply:
Ah - good call.

Reply:
Done.